### PR TITLE
Change hypothetical_check

### PR DESCRIPTION
**Pull request by huntedhippo**
Changed test none to moves instead of hypothetical moves.  Probably should just delete this altogether. No recursion implemented (yet).

Added note that hypothetical moves are not being made or respected.  Where will this be implemented?

Inverted enemy side so enemies are opposite side to king.

### CODE REVIEW

File: server/chess.py [hunk 1]
```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
                 typing.Tuple[models.Piece, int, int], ...
             ]) -> bool:
         """Check if a series of moves would put a side in check."""
-        if self.hypothetical_moves is None:
+        if moves is None:
```
Comment (Artemis21) [inline server/chess.py]:
```suggestion
        if self.hypothetical_moves is not None:
```
I forgot a `not`. The purpose of this check is to ensure that `hypothetical_check` does not call itself.

File: server/chess.py [hunk 2]
```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
...
         self.hypothetical_moves = moves    # self.get_piece will observe this
+        """***Note that the hypothetical moves are not yet being used***"""
         king = models.Piece.get(
             models.Piece.side == side,
             models.Piece.piece_type == models.PieceType.KING,
             models.Piece.game == self.game
         )
         enemies = models.Piece.select().where(
-            models.Piece.side == side,
+            models.Piece.side == side.invert,
```
Comment (Artemis21) [inline server/chess.py]:
```suggestion
            models.Piece.side == ~side,
```
`__invert__` is a "magic" method. It is not meant to be called directly, instead it is an implementation of this unary operator.

File: server/chess.py [hunk 3]
```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
                 typing.Tuple[models.Piece, int, int], ...
             ]) -> bool:
         """Check if a series of moves would put a side in check."""
-        if self.hypothetical_moves is None:
+        if moves is None:
             raise RuntimeError('Checkmate detection recursion detected.')
         self.hypothetical_moves = moves    # self.get_piece will observe this
+        """***Note that the hypothetical moves are not yet being used***"""
```
Comment (Artemis21) [inline server/chess.py]:
```suggestion
        # FIXME: Observe hypothetical moves in get_piece
```
Triple quotes should not be used for comments, the exception is docstrings which I suppose are comments.

**Review by Artemis21 — CHANGES_REQUESTED** [review]

File: server/chess.py (hunk 2 above)
Comment (huntedhippo) [reply server/chess.py]: Yes, I actually realised this later on (like 5am), and thought I'd leave it to you to explain, thanks, makes sense.

**Comment by huntedhippo** [comment]
I think I've made the changes requested.

**Review by Artemis21 — APPROVED** [review]

**Pull request merged**